src/main/java/com/nique1/musicmanagementsystem/song/SongRepository.java

```diff
@@ -0,0 +1,14 @@
+package com.nique1.musicmanagementsystem.song;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+@Repository
+public interface SongRepository extends JpaRepository<SongEntity, Integer> {
+
+    List<SongEntity> findSongsBySongId(int id);
+    List<SongEntity> findSongsByArtistName(String artistName);
+    List<SongEntity> findSongsByTrackName(String trackName);
+    List<SongEntity> findSongsByYear(int year);
```

Comment:
Jw, to wyszukiwanie trzeba by zaimplementować w inny sposób, ew. na razie sobie to odpuścić